### ISSUE TITLE
scipy/optimize/nonlin doesn't use norm_tol argument in verbose mode.

### ISSUE THREAD

**Issue by njvaughn** [scipy/scipy#10062]
Similar to bug for scipy.optimize.newton_krylov() (https://github.com/scipy/scipy/issues/4259) and scipy.opttimize.root() (https://github.com/scipy/scipy/issues/7384)
scipy/optimize/nonlin is not displaying the correct norm in verbose mode.  For example when using the Anderson rootfinder, I get the same results (and same output from verbose) when I use the default tol_norm or two other example user-defined norms.  I am not sure what norm is being used for the exit condition.  

### Reproducing code example:
```
import numpy as np
from scipy.optimize import anderson


def customNorm(residual):
    return np.sqrt(np.sum(residual**2))

def customNorm2(residual):
    return np.max(np.abs(residual))

def fixedPointFunction(x):
    """ Multiply A*x """
    A = np.array([[ 5.0, 1 ,3], 
                  [ 1, 1 ,1], 
                  [ 1, 2 ,1]])
    y = np.dot(A,x)
    y = y/np.linalg.norm(y)
    return y-x



xin = np.array([1.0, 2.0, 3.0])
print('xin = ',xin,'\n')

xin = np.array([1.0, 2.0, 3.0])
sol = anderson(fixedPointFunction,xin,verbose=True)
print(sol,'\n')
sol = anderson(fixedPointFunction,xin,tol_norm=customNorm,verbose=True)
print(sol,'\n')
sol = anderson(fixedPointFunction,xin,tol_norm=customNorm2,verbose=True)
print(sol,'\n')
```

The output, without any errors, is:

xin =  [1. 2. 3.] 

0:  |F(x)| = 1.25477; step 1; tol 0.149347
1:  |F(x)| = 0.0919624; step 1; tol 0.00483432
2:  |F(x)| = 0.1019; step 1; tol 0.9999
3:  |F(x)| = 0.00467592; step 1; tol 0.89982
4:  |F(x)| = 0.000129422; step 1; tol 0.728708
5:  |F(x)| = 2.29278e-05; step 1; tol 0.477914
6:  |F(x)| = 3.03423e-06; step 1; tol 0.205562
[0.93274114 0.23505511 0.27339296] 

0:  |F(x)| = 1.25477; step 1; tol 0.149347
1:  |F(x)| = 0.0919624; step 1; tol 0.00483432
2:  |F(x)| = 0.1019; step 1; tol 0.9999
3:  |F(x)| = 0.00467592; step 1; tol 0.89982
4:  |F(x)| = 0.000129422; step 1; tol 0.728708
5:  |F(x)| = 2.29278e-05; step 1; tol 0.477914
6:  |F(x)| = 3.03423e-06; step 1; tol 0.205562
[0.93274114 0.23505511 0.27339296] 

0:  |F(x)| = 1.25477; step 1; tol 0.149347
1:  |F(x)| = 0.0919624; step 1; tol 0.00483432
2:  |F(x)| = 0.1019; step 1; tol 0.9999
3:  |F(x)| = 0.00467592; step 1; tol 0.89982
4:  |F(x)| = 0.000129422; step 1; tol 0.728708
5:  |F(x)| = 2.29278e-05; step 1; tol 0.477914
6:  |F(x)| = 3.03423e-06; step 1; tol 0.205562
[0.93274114 0.23505511 0.27339296] 

### Scipy/Numpy/Python version information:
```
1.1.0 1.15.2 sys.version_info(major=3, minor=6, micro=8, releaselevel='final', serial=0)
```

**Comment by jconroy14** [comment]
Is this issue still present for you in the latest version of Scipy? When I run the same code (on version 1.2.1 and on 1.4.0.dev0+1953c76), I get different verbose outputs for different norm functions, as expected. I also get (slightly) different roots for different norms, suggesting that the exit condition is correctly using the user norm. I believe #9034 fixed this issue.

**Output of code in Scipy 1.2.1/Numpy 1.16.12**:
xin =  [1. 2. 3.] 

0:  |F(x)| = 1.05239; step 1
1:  |F(x)| = 0.067935; step 1
2:  |F(x)| = 0.0761608; step 1
3:  |F(x)| = 0.00381292; step 1
4:  |F(x)| = 0.000120152; step 1
5:  |F(x)| = 1.59777e-05; step 1
6:  |F(x)| = 2.53889e-06; step 1
[0.93274114 0.23505511 0.27339296] 

0:  |F(x)| = 1.25477; step 1
1:  |F(x)| = 0.0919624; step 1
2:  |F(x)| = 0.1019; step 1
3:  |F(x)| = 0.00467592; step 1
4:  |F(x)| = 0.000129422; step 1
5:  |F(x)| = 2.29278e-05; step 1
6:  |F(x)| = 3.03423e-06; step 1
[0.93274114 0.23505511 0.27339296] 

0:  |F(x)| = 1.05239; step 1
1:  |F(x)| = 0.067935; step 1
2:  |F(x)| = 0.0761608; step 1
3:  |F(x)| = 0.00381292; step 1
4:  |F(x)| = 0.000120152; step 1
5:  |F(x)| = 1.59777e-05; step 1
6:  |F(x)| = 2.53889e-06; step 1
[0.93274114 0.23505511 0.27339296]

**Comment by sethtroisi** [comment]
I get the same output as above which indicates (to me) that this was fixed by #9034.
